--- a/src/util/CADutil.cpp
+++ b/src/util/CADutil.cpp
@@ -43,7 +43,7 @@ STEPutil::STEPutil( const int & len, const double & tol )
 STEPutil::~STEPutil()
 {
     header_instances->DeleteInstances();
-    instance_list->DeleteInstances();
+  //  instance_list->DeleteInstances();
     delete registry;
     delete instance_list;
     delete sfile;
@@ -106,11 +106,11 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
             {
                 if( !strcmp( attr->Name(), "prefix" ) )
                 {
-                    attr->Raw()->e = new SdaiSi_prefix_var( pfx );
+                    attr->Enum( new SdaiSi_prefix_var( pfx ) );
                 }
                 if( !strcmp( attr->Name(), "name" ) )
                 {
-                    attr->Raw()->e = new SdaiSi_unit_name_var( Si_unit_name__metre );
+                    attr->Enum( new SdaiSi_unit_name_var( Si_unit_name__metre ) );
                 }
             }
         }
@@ -194,8 +194,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
                     }
                     if( !strcmp( attr->Name(), "conversion_factor" ) )
                     {
-                        attr->Raw()->c = new( STEPentity * );
-                        *( attr->Raw()->c ) = ( STEPentity * )( len_measure_with_unit );
+                        attr->Entity( ( STEPentity * )( len_measure_with_unit ) );
                     }
                 }
             }
@@ -206,8 +205,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
                 {
                     if( !strcmp( attr->Name(), "dimensions" ) )
                     {
-                        attr->Raw()->c = new( STEPentity * );
-                        *( attr->Raw()->c ) = ( STEPentity * )( dimensional_exp_len );
+                        attr->Entity( ( STEPentity * )( dimensional_exp_len ) );
                     }
                 }
             }
@@ -228,7 +226,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
         {
             if( !strcmp( attr->Name(), "unit_component" ) )
             {
-                attr->Raw()->sh = tol_unit;
+                attr->Select( tol_unit );
             }
             if( !strcmp( attr->Name(), "value_component" ) )
             {
@@ -256,7 +254,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
             {
                 if( !strcmp( attr->Name(), "name" ) )
                 {
-                    attr->Raw()->e = new SdaiSi_unit_name_var( Si_unit_name__radian );
+                    attr->Enum( new SdaiSi_unit_name_var( Si_unit_name__radian ) );
                 }
             }
         }
@@ -299,8 +297,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
                     }
                     if( !strcmp( attr->Name(), "conversion_factor" ) )
                     {
-                        attr->Raw()->c = new( STEPentity * );
-                        *( attr->Raw()->c ) = ( STEPentity * )( p_ang_measure_with_unit );
+                        attr->Entity( ( STEPentity * )( p_ang_measure_with_unit ) );
                     }
                 }
             }
@@ -311,8 +308,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
                 {
                     if( !strcmp( attr->Name(), "dimensions" ) )
                     {
-                        attr->Raw()->c = new( STEPentity * );
-                        *( attr->Raw()->c ) = ( STEPentity * )( dimensional_exp );
+                        attr->Entity( ( STEPentity * )( dimensional_exp ) );
                     }
                 }
             }
@@ -334,7 +330,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
             {
                 if( !strcmp( attr->Name(), "name" ) )
                 {
-                    attr->Raw()->e = new SdaiSi_unit_name_var( Si_unit_name__steradian );
+                    attr->Enum( new SdaiSi_unit_name_var( Si_unit_name__steradian ) );
                 }
             }
         }
@@ -372,7 +368,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
                 {
                     EntityAggregate * unc_agg = new EntityAggregate();
                     unc_agg->AddNode( new EntityNode( ( SDAI_Application_instance * ) uncertainty ) );
-                    attr->Raw()->a = unc_agg;
+                    attr->Aggregate( unc_agg );
                 }
             }

@@ -389,7 +385,7 @@ STEPcomplex * STEPutil::Geometric_Context( const vsp::LEN_UNITS & len, const vsp
                     unit_assigned_agg->AddNode( new EntityNode( ( SDAI_Application_instance * ) ua_length ) );
                     unit_assigned_agg->AddNode( new EntityNode( ( SDAI_Application_instance * ) ua_plane_angle ) );
                     unit_assigned_agg->AddNode( new EntityNode( ( SDAI_Application_instance * ) ua_solid_angle ) );
-                    attr->Raw()->a = unit_assigned_agg;
+                    attr->Aggregate( unit_assigned_agg );
                 }
             }
         }
